packages/cozy-harvest-lib/src/components/TwoFAModal.jsx

```diff
@@ -28,20 +30,27 @@ export class TwoFAForm extends PureComponent {
     // prevent refreshing the page when submitting
     // (happens not systematically)
     e.preventDefault()
-    this.props.konnectorJob.sendTwoFACode(this.state.twoFACode)
+    this.props.onTwoFACodeSubmit(this.state.twoFACode)
   }
 
   fetchIcon() {
     const { client } = this.context
-    const { konnectorJob } = this.props
+    const { konnectorSlug } = this.props
     return client.stackClient.getIconURL({
       type: 'konnector',
-      slug: konnectorJob.getKonnectorSlug()
+      slug: konnectorSlug
     })
```

Comment:
When the TwoFaModal is displayed, we already have the konnector icon displayed in the modal behind. So I think we could take the icon from there and display it directly in TwoFAModal. Technically, it would be better IMHO because it would remove this logic from the TwoFAModal and save a round trip to the server to fetch the icon. Functionally, it would also be better since the icon would not be reloaded : I already have seen the loading animation of the icon in the modal and tought "why is it loading ? we already have it behind" ?

Reply:
I agree. @Crash-- do you think we could integrate this change in our routing PR? We already have a component that takes care of rendering the konnector icon there, and I believe it only fetches the icon once.

Reply:
I think we can, let's create an issue. I don't want to add anything else in the upcoming PR. ATM we're unable to finish it already ^^